ceci/stage.py

```diff
@@ -72,51 +82,115 @@ def __init__(self, args, comm=None):
+
+    def load_configs(self, args):
+        """
+        Load the configuraiton
+
+        Parameters
+        ----------
+        args: dict or namespace
+            Specification of input and output paths and any missing config options
         """
         if not isinstance(args, dict):
             args = vars(args)
 
+        # First, we extract configuration information from a combination of
+        # command line arguments and optional 'config' file
+        self._inputs = dict(config=args["config"])
+        self.read_config(args)
+
         # We first check for missing input files, that's a show stopper
         missing_inputs = []
         for x in self.input_tags():
-            val = args[x]
+            val = args.get(x)
+            aliased_tag = self.get_aliased_tag(x)
+
             if val is None:
+                val = args.get(aliased_tag)
+
+            if val is None:  #pragma: no cover
                 missing_inputs.append(f"--{x}")
-        if missing_inputs:
+            else:
+                self._inputs[aliased_tag] = val
+        if missing_inputs:  #pragma: no cover
             missing_inputs = "  ".join(missing_inputs)
             raise ValueError(
                 f"""
 
-Missing these names on the command line:
+{self.instance_name} Missing these names on the command line:
     Input names: {missing_inputs}"""
             )
 
-        self._inputs = {x: args[x] for x in self.input_tags()}
         # We alwys assume the config arg exists, whether it is in input_tags or not
-        if args.get("config") is None:
+        if 'config' not in args:  #pragma: no cover
             raise ValueError("The argument --config was missing on the command line.")
 
-        self._inputs["config"] = args["config"]
-
         # We prefer to receive explicit filenames for the outputs but will
         # tolerate missing output filenames and will default to tag name in
         # current folder (this is for CWL compliance)
         self._outputs = {}
         for i, x in enumerate(self.output_tags()):
-            if args[x] is None:
+            if args.get(x) is None:
                 ftype = self.outputs[i][1]
-                self._outputs[x] = ftype.make_name(x)
+                self._outputs[self.get_aliased_tag(x)] = ftype.make_name(x)
             else:
-                self._outputs[x] = args[x]
+                self._outputs[self.get_aliased_tag(x)] = args[x]
 
-        # Finally, we extract configuration information from a combination of
-        # command line arguments and optional 'config' file
-        self._configs = self.read_config(args)
 
-        use_mpi = args.get("mpi", False)
-        if use_mpi:
+    def setup_mpi(self, comm=None):
+        """
+        Setup the MPI interface
+
+        Parameters
+        ----------
+        comm: MPI communicator
+            (default is None) An MPI comm object to use in preference to COMM_WORLD
+        """
+        use_mpi = self.config.get('use_mpi', False)
```

Comment:
This is still called `--mpi` in the code below, and I'd like to retain it as that.

Reply:
Ok, fixed.